src/client/routes/profile/ProfileView.tsx

```diff
@@ -1,45 +1,92 @@
 import * as React from "react";
-import { withRouter } from "react-router-dom";
-import TextField from "@material-ui/core/TextField";
+import { withRouter, RouteComponentProps } from "react-router-dom";
+import {TextField, WithStyles, withStyles, createStyles, Theme, Button, Typography} from "@material-ui/core";
 import ApiService from "../../common/ApiService";
 
-// TODO an actual view with some reasonable ui and more fields! email! password! whoa, very much data, so doge
-class ProfileView extends React.Component<any, any> {
-  constructor(props: any) {
+class ProfileView extends React.Component<IProfileViewProps, IProfileViewState> {
+  private submitField: any = React.createRef();
+
+  constructor(props: IProfileViewProps) {
     super(props);
+    this.state = {} as IProfileViewState
+  }
+
+  public async componentDidMount() {
+    const user = await ApiService.read.user();
+    this.setState(user);
   }
 
   public render() {
-    // todo lol login page class
+
     return (
-      <div className="login page">
-        <div className="form">
-          <h3 className="title">What's your nickname?</h3>
-          <form onSubmit={this.onSubmitLogin}>
-            <TextField name="username" label="Username" onChange={this.onInputChange} />
-          </form>
-        </div>
-      </div>
+      <form
+        className={this.props.classes.container}
+        onKeyPress={this.handleEnterKey}
+      >
+        <TextField
+          autoFocus
+          required
+          id="name"
+          label="Name"
+          className={this.props.classes.textField}
+          value={this.state.name || ""}
+          onChange={this.onInputChange}
+          margin="normal"
+        />
+        <br/>
+        <TextField
+          required
+          id="email"
+          label="Email"
+          className={this.props.classes.textField}
+          value={this.state.email || ""}
+          onChange={this.onInputChange}
+          margin="normal"
+          inputRef={this.submitField}
+        />
+        <br/>
+        <Button
+          variant="raised"
+          color="primary"
+          className={this.props.classes.button}
+          onClick={this.handleSubmit}
+        >
+          <Typography color="inherit">Save</Typography>
+        </Button>
+      </form>
     )
   }
 
   public onInputChange = ({target}) => {
-      this.setState({[target.name]: target.value});
+    this.setState({...this.state, [target.id]: target.value});
   }
 
-  public onSubmitLogin = async (event) => {
-    event.preventDefault();
-    const name = this.cleanInput(this.state.username);
-    if (name) {
-      // Add separate button to navigate away from profile view? "back to lobby"
-      await ApiService.write.user({name} as global.IUserContract);
+  public handleSubmit = async () => {
+    if (this.state.name && this.state.email) {
+      await ApiService.write.user(this.state);

```

Comment:
Probably better to send only the wanted information to Api, and not the whole state.